src/Parsers.jl

```diff
@@ -33,81 +33,131 @@ struct Result{T}
     Result{T}(code::ReturnCode, tlen::Integer, val) where {T} = new{T}(code, tlen, val)
 end
 
-# Result(code::ReturnCode, tlen::Integer, val::T) where {T} = Result{T}(code, tlen, val)
+Base.show(io::IO, x::Result{T}) where {T} = print(io, "Parsers.Result{$T}(code=`", codes(x.code), "`, tlen=", x.tlen, ", val=", isdefined(x, :val) ? x.val : "#undefined", ")")
+
+# bit flags to hold several Bool options for Options
+primitive type Flags 8 end
+
+const WHDELIM         = 0b00000001
+const STRIPQUOTED     = 0b00000010
+const STRIPWHITESPACE = 0b00000100
+const CHECKQUOTED     = 0b00001000
+const CHECKSENTINEL   = 0b00010000
+const CHECKDELIM      = 0b00100000
+const IGNOREREPEATED  = 0b01000000
+const IGNOREEMPTY     = 0b10000000
+
+function Flags(whitespacedelim, stripquoted, stripwhitespace, checkquoted, checksentinel, checkdelim, ignorerepeated, ignoreemptylines)
+    x = 0x00
+    whitespacedelim && (x |= WHDELIM)
+    stripquoted && (x |= STRIPQUOTED)
+    # stripquoted implies stripwhitespace
+    (stripwhitespace || stripquoted) && (x |= STRIPWHITESPACE)
+    checkquoted && (x |= CHECKQUOTED)
+    checksentinel && (x |= CHECKSENTINEL)
+    checkdelim && (x |= CHECKDELIM)
+    ignorerepeated && (x |= IGNOREREPEATED)
+    ignoreemptylines && (x |= IGNOREEMPTY)
+    return Base.bitcast(Flags, x)
+end
+
+function Base.getproperty(x::Flags, nm::Symbol)
```

Comment:
should we add an overload for `Options` too? otherwise users need to change e.g. `options.ignorerepeated` -> `options.flags.ignorerepeated` (i.e. this is a breaking change)

Reply:
maybe we can just forward unknown names to this `Flags` method? e.g.
```julia
function Base.getproperty(opts::Options, nm::Symbol)
    return hasfield(Options, nm) ? getfield(opts, nm) : getproperty(opts.flags, nm)
end
```

Reply:
Yeah, see the `getproperty` for `Options` a little further down where I implemented this.